app/components/collapse_control_component.html.erb

```diff
@@ -1,7 +1,22 @@
-<%= tag.button type: "button", data: {action: "toggle#toggle accessibility#toggleAriaExpanded", accessibility_target: "button"}, "aria-expanded": !collapsed?, class: "px-2 py-3 w-full flex items-center justify-between text-gray-400 hover:text-gray-500 text-sm" do %>
-  <%= tag.span title, class: ["font-medium text-gray-900", "text-xs": subcomponent?] %>
-  <span class="ml-6 flex items-center">
-    <%= inline_svg_tag "icons/solid/minus_sm.svg", class: minus_icon_class, data: {"toggle-target": "element"}, title: "Open menu" %>
-    <%= inline_svg_tag "icons/solid/plus_sm.svg", class: plus_icon_class, data: {"toggle-target": "element"}, title: "Close menu" %>
-  </span>
-<% end %>
+<div class="flex items-center">
+  <%= tag.button type: "button", data: {action: "toggle#toggle accessibility#toggleAriaExpanded", accessibility_target: "button"}, "aria-expanded": !collapsed?, class: "px-2 py-3 w-full flex items-center justify-between text-gray-400 hover:text-gray-500 text-sm" do %>
+    <%= tag.span title, class: ["font-medium text-gray-900", "text-xs": subcomponent?] %>
+    <% if paywalled? %>
+      <% if customer? %>
+        <span class="ml-6 flex items-center">
+          <%= inline_svg_tag "icons/solid/minus_sm.svg", class: minus_icon_class, data: {"toggle-target": "element"}, title: "Open menu" %>
+          <%= inline_svg_tag "icons/solid/plus_sm.svg", class: plus_icon_class, data: {"toggle-target": "element"}, title: "Close menu" %>
+        </span>
+      <% end %>
+    <% else %>
+      <span class="ml-6 flex items-center">
+        <%= inline_svg_tag "icons/solid/minus_sm.svg", class: minus_icon_class, data: {"toggle-target": "element"}, title: "Open menu" %>
+        <%= inline_svg_tag "icons/solid/plus_sm.svg", class: plus_icon_class, data: {"toggle-target": "element"}, title: "Close menu" %>
+      </span>
+    <% end %>
+  <% end %>
+
+  <% if paywalled? && !customer? %>
+    <%= render Users::PaywalledComponent.new(user: user, paywalled: nil, size: :smallest) %>
+  <% end %>
```

Comment:
I don't like mixing logic of paywalls + customers with a generic control like this. Instead, I think we should refactor `Users::PaywalledComponent` to accept a slot for custom CTA content.

Reply:
Yeah, I don't like that either. Using a slot to handle custom CTA content is a nice idea